### PR TITLE
Make client authentication configurable

### PR DESCRIPTION
**Pull request by hugo19941994**
**Description of the change**

This change adds a new environment variable (`KAFKA_CFG_SSL_CLIENT_AUTH`) which allows users to choose whether to enable client authentication or not, while keeping encryption enabled. This new environment variable maps 1 to 1 with Kafka's `ssl.client.auth`. As such, the same values can be used:

* `none`: Disables client authentication completely
* `requested`: Enables client authentication without making it mandatory
* `required`: Enables client authentication for all client connections (default value to keep backwards compatibility)

If an invalid value is used an error message is logged indicating which values should be used.

To maintain backwards compatibility, `required` is the default value (which was previously hardcoded).

**Benefits**

Adds flexibility when deciding how to secure the Kafka cluster. While mTLS provides both encryption and authentication, in some instances it can be a hassle managing certificates for clients. With this option brokers can be deployed with TLS enabled, which provides encryption without the burden of generating client certificates.

**Possible drawbacks**

**Applicable issues**

**Additional information**

[Link to Kafka ssl.client.auth docs](https://kafka.apache.org/documentation/#connectconfigs_ssl.client.auth)

### CODE REVIEW

File: 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 1]
```diff
@@ -287,6 +288,9 @@ kafka_validate() {
     elif ! is_boolean_yes "$ALLOW_PLAINTEXT_LISTENER"; then
          print_validation_error "The KAFKA_ZOOKEEPER_PROTOCOL environment variable does not configure a secure protocol. Set the environment variable ALLOW_PLAINTEXT_LISTENER=yes to allow the container to be started with a plaintext listener. This is only recommended for development."
     fi
+    if [[ "${KAFKA_CFG_SSL_CLIENT_AUTH}" != "none" ]] && [[ "${KAFKA_CFG_SSL_CLIENT_AUTH}" != "requested" ]] && [[ "${KAFKA_CFG_SSL_CLIENT_AUTH}" != "required" ]]; then
+        print_validation_error "The KAFKA_CFG_SSL_CLIENT_AUTH environment variable must be one of the following: none, requested or required"
+    fi
```
Comment (migruiz4) [inline 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh:293]: To maintain consistency with other Bitnami images, could you please use instead the function `check_multi_values()`?

```sh
check_multi_value "KAFKA_CFG_SSL_CLIENT_AUTH" "none requested required"
```

The function hasn't been added to this file yet, could you please add it using this file as a reference?
https://github.com/bitnami/bitnami-docker-joomla/blob/24f9ab84f7e50ff08fc22e6bd5f96d24a56b7dfd/3/debian-10/rootfs/opt/bitnami/scripts/libjoomla.sh#L47-L51

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! Could you please have a look at the suggestion I added?

File: 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh (hunk 1 above)
Comment (hugo19941994) [reply 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh]: Changed

**Review by migruiz4 — APPROVED** [review]
Looks good to me, thanks!

File: 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 2]
```diff
@@ -296,6 +297,7 @@ kafka_validate() {
     elif ! is_boolean_yes "$ALLOW_PLAINTEXT_LISTENER"; then
          print_validation_error "The KAFKA_ZOOKEEPER_PROTOCOL environment variable does not configure a secure protocol. Set the environment variable ALLOW_PLAINTEXT_LISTENER=yes to allow the container to be started with a plaintext listener. This is only recommended for development."
     fi
+    check_multi_value "KAFKA_CFG_SSL_CLIENT_AUTH" "none requested required"
```
Comment (juan131) [inline 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh:300]: You're treating the 3 words as a single string:
```suggestion
    check_multi_value "KAFKA_CFG_SSL_CLIENT_AUTH" "none" "requested" "required"
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for your PR! Please check my comments.

File: 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 3]
```diff
@@ -124,6 +124,7 @@ export KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE="${KAFKA_CFG_AUTO_CREATE_TOPICS_ENABL
 export KAFKA_CFG_SASL_ENABLED_MECHANISMS="${KAFKA_CFG_SASL_ENABLED_MECHANISMS:-PLAIN,SCRAM-SHA-256,SCRAM-SHA-512}"
 export KAFKA_CFG_SASL_MECHANISM_INTER_BROKER_PROTOCOL="${KAFKA_CFG_SASL_MECHANISM_INTER_BROKER_PROTOCOL:-}"
 export KAFKA_CFG_TLS_TYPE="${KAFKA_CFG_TLS_TYPE:-JKS}"
+export KAFKA_CFG_SSL_CLIENT_AUTH="${KAFKA_CFG_SSL_CLIENT_AUTH:-required}"
```
[the remark this answers is not in the document]
Comment (hugo19941994) [reply 2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh]: Done

**Review by juan131 — APPROVED** [review]
Thanks for implementing the suggestions! LGTM

**Comment by juan131** [comment]
(sending your PR for internal review, I'll keep you updated)

**Comment by juan131** [comment]
PR accepted in internal review, merging your PR!
What's next? A new image revision will be released including your changes. I'll keep you updated once it's released.

**Comment by juan131** [comment]
@hugo19941994 we just released the revision **2.7.0-debian-10-r40** including your changes. Please give it a try when you have a chance.

**Pull request merged**